crates/nix_rs/src/flake/schema.rs

```diff
@@ -1,108 +1,216 @@
+                        .collect();
+                    if filtered.is_empty() {
+                        None
+                    } else {
+                        Some(FlakeOutputs::Attrset(filtered))
                     }
                 }
-                item.pop(ks);
             }
+            _ => None,
+        }
+    }
+}
 
-            result
-        };
-        let pop_per_system_tree = |output: &mut FlakeOutputs, k: &str| -> BTreeMap<String, Leaf> {
-            pop_tree(
-                &mut output.inventory.get_mut(k),
-                &["children", system.as_ref(), "children"],
-            )
-        };
-        let pop_leaf_type = |output: &mut FlakeOutputs, k: &str| -> Option<Leaf> {
-            let inventory_item = output.inventory.get_mut(k)?;
-            let leaf = inventory_item
-                .pop(&["children", system.as_ref()])?
-                .as_leaf()?
-                .clone();
-            inventory_item.pop(&[k]);
-            Some(leaf)
-        };
+/// Represents a leaf value of a flake output
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Leaf {
+    Val(Val),
+    Unknown(Unknown),
+    Filtered(Filtered),
+    Skipped(Skipped),
+    /// Represents description for a flake output
+    /// (e.g. `Doc` for `formatter` will be "The `formatter` output specifies the package to use to format the project.")
+    Doc(String),
+}
 
-        FlakeSchema {
-            system: system.clone(),
-            packages: pop_per_system_tree(output, "packages"),
-            legacy_packages: pop_per_system_tree(output, "legacyPackages"),
-            devshells: pop_per_system_tree(output, "devShells"),
-            checks: pop_per_system_tree(output, "checks"),
-            apps: pop_per_system_tree(output, "apps"),
-            formatter: pop_leaf_type(output, "formatter"),
-            nixos_configurations: pop_tree(
-                &mut output.inventory.get_mut("nixosConfigurations"),
-                &["children"],
-            ),
-            darwin_configurations: pop_tree(
-                &mut output.inventory.get_mut("darwinConfigurations"),
-                &["children"],
-            ),
-            home_configurations: pop_tree(
-                &mut output.inventory.get_mut("homeConfigurations"),
-                &["children"],
-            ),
-            nixos_modules: pop_tree(&mut output.inventory.get_mut("nixosModules"), &["children"]),
-            docker_images: pop_tree(&mut output.inventory.get_mut("dockerImages"), &["children"]),
-            overlays: pop_tree(&mut output.inventory.get_mut("overlays"), &["children"]),
-            templates: pop_tree(&mut output.inventory.get_mut("templates"), &["children"]),
-            schemas: pop_tree(&mut output.inventory.get_mut("schemas"), &["children"]),
-            other: Some(output.clone()),
+/// The metadata of a flake output value which is of non-attrset [Type]
```

Comment:
Might as well plainly call it "terminal value" (no children) instead of leverage understanding of Rust types.

Reply:
https://github.com/juspay/omnix/pull/235/commits/e5c6d6a46b31148d2f44350e9e2275d23eb53036